crates/ide-diagnostics/src/handlers/non_exhaustive_let.rs

```diff
@@ -41,6 +41,27 @@ fn main() {
 fn main() {
     let Some(_) | None = Some(5);
 }
+"#,
+        );
+    }
+
+    #[test]
+    fn min_exhaustive() {
+        check_diagnostics(
+            r#"
+//- minicore: result
+fn test(x: Result<i32, !>) {
+    let Ok(_y) = x;
+}
+"#,
+        );
```

Comment:
The reason that this test doesn't have a pointer deref case is because the following code;
```rust
fn test(ptr: *const Result<i32, !>) {
    unsafe {
        let Ok(_x) = *ptr;
    }
}
```
is getting a block with no stmts but tail one in here,
https://github.com/rust-lang/rust-analyzer/blob/0daeb5c0b05cfdf2101b0f078c27539099bf38e6/crates/hir-ty/src/diagnostics/expr.rs#L256-L257
while the following is getting a block with a single stmt without tail 🤔 
```rust
fn test(x: Result<i32, &'static !>) {
    let Ok(_y) = x;
}
```
I'll make a more deep inspection and issue this as a new issue

Reply:
> is getting a block with no stmts but tail one in here(thus, no diagnostic error),

oh no that sounds very wrong

Reply:
Yes, it looks very wrong 🤔 I'll check the entire lowering process from AST to hir once I get back home

Reply:
This is because #17865

Reply:
I got into this 😂 rust-lang/rust#129009